File: library-core/src/main/java/com/mikepenz/fastadapter/listeners/ItemFilterListener.java
package com.mikepenz.fastadapter.listeners;

import com.mikepenz.fastadapter.IItem;

import java.util.List;

import javax.annotation.Nullable;

/**
 * interface for the ItemFilterListener
 */
public interface ItemFilterListener<Item extends IItem> {
    void itemsFiltered(@Nullable CharSequence constraint, @Nullable List<Item> results);

    void onReset();
}

File: library-core/src/main/java/com/mikepenz/fastadapter/listeners/ItemFilterListener.kt
package com.mikepenz.fastadapter.listeners

import com.mikepenz.fastadapter.GenericItem

/**
 * Interface for the ItemFilterListener
 */
interface ItemFilterListener<Item : GenericItem> {
    fun itemsFiltered(constraint: CharSequence?, results: List<Item>?)

    fun onReset()
}
